include/xtensor/xchunked_view.hpp

```diff
@@ -17,38 +17,51 @@ namespace xt
 {
 
     template <class E>
-    class xchunk_generator
+    class xchunk_iterator
     {
     public:
         template <class OE>
-        xchunk_generator(OE&& e, std::vector<std::size_t>& shape, std::vector<std::size_t>& chunk_shape);
+        xchunk_iterator(OE&& e, std::vector<std::size_t>& shape, std::vector<std::size_t>& chunk_shape);
 
-        xstrided_slice_vector& next();
-        bool last_chunk();
-        xstrided_slice_vector get_slice();
-        auto get_chunk();
-
-        template <class OE>
-        xchunk_generator<E>& operator=(const OE& e);
+        xchunk_iterator<E> begin();
+        xchunk_iterator<E> end();
+        xchunk_iterator<E>& operator++();
+        xchunk_iterator<E> operator++(int);
+        bool operator==(const xchunk_iterator& other) const;
+        bool operator!=(const xchunk_iterator& other) const;
+        auto operator*();
 
     private:
+        E m_expression;
         std::vector<std::size_t> m_shape;
         std::vector<std::size_t> m_chunk_shape;
-        E m_expression;
-        bool m_last_chunk;
         std::vector<std::size_t> m_shape_of_chunks;
         std::vector<std::size_t> m_ic;
-        xstrided_slice_vector m_sv;
         std::size_t m_ci;
         std::size_t m_chunk_nb;
+        xstrided_slice_vector m_sv;
```

Comment:
All these members but `m_ci` should be in `xchunked_view`. The iterator must be as light as possible to avoid degrading performances.

The iterator class can be declared as a friend class of `xchinked_view` if it needs to access other members.